edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java

```diff
@@ -98,7 +99,25 @@
 
     private Authz authz;
     private GradebookPermissionService gradebookPermissionService;
+    private GradebookFrameworkService gradebookFrameworkService;
+    
     protected SiteService siteService;
+
+    public Gradebook addOrGetGradebook(String uid) throws GradebookNotFoundException {
```

Comment:
This is a bit circular, both GradebookFrameworkService and GradebookServiceHibernateImpl extend BaseHibernateManagerand now one is injected into the other.